src/Main.elm

```diff
@@ -49,126 +49,200 @@ type DecimalStatus
-        model
-        (model.valueA * model.valueB)
+        ( Just a, Nothing ) ->
+            checkActionAandNewAction model a newAction
 
+        ( Just a, Just b ) ->
+            checkActionBandNewAction model a b newAction
 
-executeDivide : Model -> Model
-executeDivide model =
-    saveResults
-        model
-        (model.valueA / model.valueB)
+        ( Nothing, Just b ) ->
+            model
 
 
-showResults : Float -> String
-showResults float =
-    let
-        string =
-            String.fromFloat float
+saveActionA : Model -> Action -> Model
+saveActionA model action =
+    if action == Equals then
+        model
 
-        hasDecimal =
-            String.any (\a -> '.' == a) string
+    else
+        { model
+            | valueA = displatToFloat model.display
+            , actionA = Just action
+            , displayStatus = ShowingResults
+        }
 
-        largerThanTen =
-            if hasDecimal then
-                String.length string > 11
 
-            else
-                String.length string > 10
-    in
-    if largerThanTen then
-        "ERROR"
+actionTypeOf : Action -> ActionType
+actionTypeOf action =
+    case action of
+        Add ->
+            AddSubtractType
 
-    else
-        string
+        Subtract ->
+            AddSubtractType
 
+        Multiply ->
+            MultiplyDivideType
 
+        Divide ->
+            MultiplyDivideType
 
--- Update
+        Equals ->
+            EqualsType
 
 
-type Msg
-    = UpdateDisplay CalButton
-    | Saveinput Action
-    | AllClear
+checkActionAandNewAction : Model -> Action -> Action -> Model
+checkActionAandNewAction model actionA newAction =
+    if
+        (actionTypeOf actionA == actionTypeOf newAction)
+            || (newAction == Equals)
+    then
+        doMathOnAandNewAction model actionA newAction
+
+    else
+        saveActionB model newAction
 
 
-savedValueBs : Model -> Action -> Model
-savedValueBs model action =
+saveActionB : Model -> Action -> Model
+saveActionB model action =
     { model
-        | valueB = displatToFloat model.display
+        | displayStatus = ShowingResults
+        , valueB = displatToFloat model.display
         , actionB = Just action
+    }
+
+
+doMathOnAandNewAction : Model -> Action -> Action -> Model
+doMathOnAandNewAction model actionA newAction =
+    let
+        results =
+            simpleMath
+                model.valueA
+                actionA
+                (displatToFloat model.display)
+    in
+    { model
+        | display = showResults <| results
         , displayStatus = ShowingResults
+        , valueA = results
+        , actionA = Just newAction
+        , valueB = 0
+        , actionB = Nothing
     }
 
 
-saveAction : Model -> Action -> Model
-saveAction model action =
-    case model.actionA of
-        Nothing ->
-            if action == Equals then
-                model
+simpleMath : Float -> Action -> Float -> Float
+simpleMath numOne action numTwo =
```

Comment:
This is a nice, well expressed function with the appropriate types. :+1: :heart: